internal/util/tags/servertags/join.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2024 The ORC Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servertags
+
+import (
+	"context"
+	"strings"
+
+	"k8s.io/utils/set"
+
+	"github.com/gophercloud/gophercloud/v2/openstack/compute/v2/tags"
+	orcv1alpha1 "github.com/k-orc/openstack-resource-controller/v2/api/v1alpha1"
+	"github.com/k-orc/openstack-resource-controller/v2/internal/controllers/generic/interfaces"
+	"github.com/k-orc/openstack-resource-controller/v2/internal/controllers/generic/progress"
+	"github.com/k-orc/openstack-resource-controller/v2/internal/osclients"
+)
+
+// Join joins a slice of tags into a comma separated list of tags.
+func Join(tags []orcv1alpha1.ServerTag) string {
+	var b strings.Builder
+	for i := range tags {
+		if i > 0 {
+			b.WriteString(",")
+		}
+		b.WriteString(string(tags[i]))
+	}
+	return b.String()
+}
+
+func ReconcileTags[orcObjectPT, osResourceT any](
```

Comment:
If the implementation doesn't differ from the neutron tags, can we consider using generics to have a common implementation for all resources?